### PR TITLE
Drop support for legacy Python 2.7

### PR DESCRIPTION
**Pull request by hugovk**
Fixes https://github.com/googlefonts/glyphsLib/issues/544.

### CODE REVIEW

**Comment by madig** [comment]
Uh-oh, I feared this when I saw you forking this repo but had no way of contacting you. I haven't actually heard back from @anthrotype if we should drop Python 2 support already. I know I want to, but this lib can theoretically be used in Glyphs, which is Py2 only. So... @anthrotype? We want to heavily refactor the lib anyway at some point in the undefined future. Hmhmhm.

File: Lib/glyphsLib/classes.py [hunk 1]
```diff
@@ -38,7 +36,8 @@
 from glyphsLib.parser import Parser
 from glyphsLib.writer import Writer
 from collections import OrderedDict
-from fontTools.misc.py23 import unicode, basestring, UnicodeIO, unichr, open
+from io import StringIO
+from fontTools.misc.py23 import unicode
```
Comment (anthrotype) [inline Lib/glyphsLib/classes.py]: you probably forgot to remove this line

**Comment by anthrotype** [comment]
> this lib can theoretically be used in Glyphs, which is Py2 only

But do people actually use glyphsLib from within the Glyphs.app Macros window? Why would they want to use it? If it's for exporting/importing to UFO then they can keep using it from the command-line. 
I spoke with @schriftgestalt in Tokyo and he said that python3 support may be coming soon for Glyphs.app.

**Comment by madig** [comment]
Continue at https://github.com/googlefonts/glyphsLib/pull/554.

**Pull request closed without merging**